org.eclipse.xtext.xbase.lib.tests/src/org/eclipse/xtext/xbase/tests/lib/IterableExtensionsTest.java

```diff
@@ -194,6 +195,39 @@ public String apply(String p) {
 		assertEquals(new Pair<Integer, String>(1, "bar"), result.next());
 		assertFalse(result.hasNext());
 	}
-
 	
+	@Test public void testReject() {
+		Function1<Integer, Boolean> function = new Function1<Integer, Boolean>() {
+			@Override
+			public Boolean apply(Integer p) {
+				return p % 2 == 0;
+			}
+		};
+		assertEquals(newArrayList(1,3,5),newArrayList(IterableExtensions.reject(newArrayList(1,2,3,4,5), function)));
+		try {
+			newArrayList(IterableExtensions.reject(null, function));
+			fail("NullPointerException expected");
+		} catch (NullPointerException e) {
+			// expected NPE
+		}
+		try {
+			newArrayList(IterableExtensions.reject(newArrayList(1,2,3), null));
+			fail("NullPointerException expected");
+		} catch (NullPointerException e) {
+			// expected NPE
+		}
+		Function1<Integer, Boolean> brokenFunction = new Function1<Integer, Boolean>() {
+			@Override
+			public Boolean apply(Integer p) {
+				return null;
+			}
+		};
+		try {
+			newArrayList(IterableExtensions.reject(newArrayList(1,2,3), brokenFunction));
+			fail("NullPointerException expected");
```

Comment:
this handles `myStrings.reject[null]` - same as iterables not filter does not allow this